frame/benchmarking/src/lib.rs

```diff
@@ -1328,8 +1328,20 @@ macro_rules! add_benchmark {
 			repeat,
 			verify,
 			extra,
+			list,
 		} = config;
-		if &pallet[..] == &name_string[..] || &pallet[..] == &b"*"[..] {
+
+		if *list {
+			$crate::log::info!("\npallet: {}", stringify!($name));
+			for benchmark in $( $location )*::benchmarks(*extra).into_iter() {
+				$crate::log::info!(
+					"benchmarks: {}",
+						$crate::sp_std::str::from_utf8(benchmark)
+							.map_err(|_| "`extrinsic` is not a valid utf8 string!")?
+				);
+			}
```

Comment:
I am not a fan of this solution. Here you are using runtime logging and also making a specific function do double duty.

I think that we should separate the list function from the `add_benchmark!` macro.

Reply:
definitely can get to print from client :+1:.  So would you want something along the lines of this in `add_benchmark!`
```rust
if *list {
        list($name, $( $location )* ))
}
```

Or would you rather adding the `benchmark_list!` directly to runtime as not to alter `add_benchmark!` at all

Reply:
I think probably we need to split `add_benchmark!`. One macro which builds the list of available benchmarks, and then another function which executes the selected benchmarks.

Then you would plug your logic as an alternative logical path instead of running the benchmarks just to list them.

Something like that?